selectrum.el

```diff
@@ -641,15 +646,21 @@ INPUT defaults to current selectrum input string."
       (completion-metadata-get
        (completion-metadata input table pred) setting))))
 
-(defun selectrum-exhibit ()
-  "Trigger an update of Selectrum's completion UI."
+(defun selectrum-exhibit (&optional keep-selection)
+  "Trigger an update of Selectrum's completion UI.
+If KEEP-SELECTION is non-nil keep the current candidate selected
+when possible."
   (when-let ((mini (active-minibuffer-window)))
     (with-selected-window mini
       (when (and minibuffer-completion-table
                  (not selectrum--dynamic-candidates))
         (setq selectrum--preprocessed-candidates nil))
       (setq selectrum--previous-input-string nil)
-      (selectrum--minibuffer-post-command-hook))))
+      (let ((selectrum--keep-candidate
+             (if keep-selection
+                 (selectrum-get-current-candidate)
+               selectrum--keep-candidate)))
+        (selectrum--minibuffer-post-command-hook)))))
```

Comment:
why not call the hook here with the current candidate to keep as argument? Maybe the selectrum--minibuffer-post-command-hook should be extracted into a separate selectrum--post-command function which is then called from both here and from the selectrum--minibuffer-post-command-hook. I don't think it is the best style to call hook functions from somewhere else, they should only be called via run-hooks after putting them in the appropriate hook list.

Reply:
I followed your suggestions, I also don't want to add an argument to the hook function itself. Could you test this with your async function?